### PR TITLE
Add support for getting datasource names from types.db

### PR DESCRIPTION
**Pull request by hasso**

### CODE REVIEW

**Comment by octo** [comment]
LGTM, thanks @hasso!

File: main.go [hunk 1]
```diff
@@ -237,6 +238,18 @@ func startCollectdServer(ctx context.Context, w api.Writer) {
 		srv.PasswordLookup = network.NewAuthFile(*collectdAuth)
 	}
 
+	if *collectdTypesDB != "" {
+		file, err := os.Open(*collectdTypesDB)
+		if err != nil {
+			log.Fatalf("Can't open types.db file %s", *collectdTypesDB)
+		}
+		typesDB, err := api.NewTypesDB(file)
```
Comment (juliusv) [inline main.go]: Close `file` after parsing it here.

**Comment by juliusv** [comment]
:+1: otherwise, thanks!

File: main.go (hunk 1 above)
Comment (hasso) [inline main.go]: Oops, indeed, fixed.

**Comment by fabxc** [comment]
Thanks

**Pull request merged**